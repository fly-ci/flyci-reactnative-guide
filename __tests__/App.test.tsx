/**
 * @format
 */

import 'react-native';
import React from 'react';
import { render, screen } from '@testing-library/react-native';
import { it, expect } from '@jest/globals';
import App from '../App';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  renderer.create(<App />);
});

it('should render content correctly', () => {
  const appContent = renderer.create(<App />).toJSON();

  expect(appContent).toMatchSnapshot();
});

it('should render Step One', async () => {
  render(<App />);

  const stepOne = await screen.findByText('Step One');

  expect(stepOne).toHaveTextContent('Step Two');
});
