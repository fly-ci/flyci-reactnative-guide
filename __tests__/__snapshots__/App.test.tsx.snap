// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render content correctly 1`] = `
<RCTSafeAreaView
  style={
    {
      "backgroundColor": "#F3F3F3",
    }
  }
>
  <RCTScrollView
    contentInsetAdjustmentBehavior="automatic"
    style={
      {
        "backgroundColor": "#F3F3F3",
      }
    }
  >
    <View>
      <View
        accessibilityIgnoresInvertColors={true}
        style={
          [
            {
              "paddingBottom": 40,
              "paddingHorizontal": 32,
              "paddingTop": 96,
            },
            {
              "backgroundColor": "#F3F3F3",
            },
          ]
        }
      >
        <Image
          accessibilityRole="image"
          source={
            {
              "testUri": "../Libraries/NewAppScreen/components/logo.png",
            }
          }
          style={
            [
              {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              },
              {
                "height": undefined,
                "width": undefined,
              },
              {
                "marginBottom": -192,
                "marginLeft": -128,
                "opacity": 0.2,
                "overflow": "visible",
                "resizeMode": "cover",
              },
            ]
          }
          testID="new-app-screen-header"
        />
        <Text
          style={
            [
              {
                "fontSize": 40,
                "fontWeight": "700",
                "textAlign": "center",
              },
              {
                "color": "#000",
              },
            ]
          }
        >
          Welcome to
          

          React Native
        </Text>
      </View>
      <View
        style={
          {
            "backgroundColor": "#FFF",
          }
        }
      >
        <View
          style={
            {
              "marginTop": 32,
              "paddingHorizontal": 24,
            }
          }
        >
          <Text
            style={
              [
                {
                  "fontSize": 24,
                  "fontWeight": "600",
                },
                {
                  "color": "#000",
                },
              ]
            }
          >
            Step One
          </Text>
          <Text
            style={
              [
                {
                  "fontSize": 18,
                  "fontWeight": "400",
                  "marginTop": 8,
                },
                {
                  "color": "#444",
                },
              ]
            }
          >
            Edit 
            <Text
              style={
                {
                  "fontWeight": "700",
                }
              }
            >
              App.tsx
            </Text>
             to change this screen and then come back to see your edits.
          </Text>
        </View>
        <View
          style={
            {
              "marginTop": 32,
              "paddingHorizontal": 24,
            }
          }
        >
          <Text
            style={
              [
                {
                  "fontSize": 24,
                  "fontWeight": "600",
                },
                {
                  "color": "#000",
                },
              ]
            }
          >
            See Your Changes
          </Text>
          <Text
            style={
              [
                {
                  "fontSize": 18,
                  "fontWeight": "400",
                  "marginTop": 8,
                },
                {
                  "color": "#444",
                },
              ]
            }
          >
            <Text>
              Press 
              <Text
                style={
                  {
                    "fontWeight": "700",
                  }
                }
              >
                Cmd + R
              </Text>
               in the simulator to reload your app's code.
            </Text>
          </Text>
        </View>
        <View
          style={
            {
              "marginTop": 32,
              "paddingHorizontal": 24,
            }
          }
        >
          <Text
            style={
              [
                {
                  "fontSize": 24,
                  "fontWeight": "600",
                },
                {
                  "color": "#000",
                },
              ]
            }
          >
            Debug
          </Text>
          <Text
            style={
              [
                {
                  "fontSize": 18,
                  "fontWeight": "400",
                  "marginTop": 8,
                },
                {
                  "color": "#444",
                },
              ]
            }
          >
            <Text>
              Press 
              <Text
                style={
                  {
                    "fontWeight": "700",
                  }
                }
              >
                Cmd + D
              </Text>
               in the simulator or
               
              <Text
                style={
                  {
                    "fontWeight": "700",
                  }
                }
              >
                Shake
              </Text>
               your device to open the Dev Menu.
            </Text>
          </Text>
        </View>
        <View
          style={
            {
              "marginTop": 32,
              "paddingHorizontal": 24,
            }
          }
        >
          <Text
            style={
              [
                {
                  "fontSize": 24,
                  "fontWeight": "600",
                },
                {
                  "color": "#000",
                },
              ]
            }
          >
            Learn More
          </Text>
          <Text
            style={
              [
                {
                  "fontSize": 18,
                  "fontWeight": "400",
                  "marginTop": 8,
                },
                {
                  "color": "#444",
                },
              ]
            }
          >
            Read the docs to discover what to do next:
          </Text>
        </View>
        <View
          style={
            {
              "marginTop": 32,
              "paddingHorizontal": 24,
            }
          }
        >
          <View
            style={
              [
                {
                  "height": 0.5,
                },
                {
                  "backgroundColor": "#DAE1E7",
                },
              ]
            }
          />
          <View
            accessibilityRole="button"
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "alignItems": "center",
                "flexDirection": "row",
                "flexWrap": "wrap",
                "justifyContent": "space-between",
                "opacity": 1,
                "paddingVertical": 8,
              }
            }
          >
            <Text
              style={
                {
                  "color": "#1292B4",
                  "flex": 2,
                  "fontSize": 18,
                  "fontWeight": "400",
                }
              }
            >
              The Basics
            </Text>
            <Text
              style={
                [
                  {
                    "flex": 3,
                    "fontSize": 18,
                    "fontWeight": "400",
                    "paddingVertical": 16,
                  },
                  {
                    "color": "#444",
                  },
                ]
              }
            >
              Explains a Hello World for React Native.
            </Text>
          </View>
          <View
            style={
              [
                {
                  "height": 0.5,
                },
                {
                  "backgroundColor": "#DAE1E7",
                },
              ]
            }
          />
          <View
            accessibilityRole="button"
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "alignItems": "center",
                "flexDirection": "row",
                "flexWrap": "wrap",
                "justifyContent": "space-between",
                "opacity": 1,
                "paddingVertical": 8,
              }
            }
          >
            <Text
              style={
                {
                  "color": "#1292B4",
                  "flex": 2,
                  "fontSize": 18,
                  "fontWeight": "400",
                }
              }
            >
              Style
            </Text>
            <Text
              style={
                [
                  {
                    "flex": 3,
                    "fontSize": 18,
                    "fontWeight": "400",
                    "paddingVertical": 16,
                  },
                  {
                    "color": "#444",
                  },
                ]
              }
            >
              Covers how to use the prop named style which controls the visuals.
            </Text>
          </View>
          <View
            style={
              [
                {
                  "height": 0.5,
                },
                {
                  "backgroundColor": "#DAE1E7",
                },
              ]
            }
          />
          <View
            accessibilityRole="button"
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "alignItems": "center",
                "flexDirection": "row",
                "flexWrap": "wrap",
                "justifyContent": "space-between",
                "opacity": 1,
                "paddingVertical": 8,
              }
            }
          >
            <Text
              style={
                {
                  "color": "#1292B4",
                  "flex": 2,
                  "fontSize": 18,
                  "fontWeight": "400",
                }
              }
            >
              Layout
            </Text>
            <Text
              style={
                [
                  {
                    "flex": 3,
                    "fontSize": 18,
                    "fontWeight": "400",
                    "paddingVertical": 16,
                  },
                  {
                    "color": "#444",
                  },
                ]
              }
            >
              React Native uses flexbox for layout, learn how it works.
            </Text>
          </View>
          <View
            style={
              [
                {
                  "height": 0.5,
                },
                {
                  "backgroundColor": "#DAE1E7",
                },
              ]
            }
          />
          <View
            accessibilityRole="button"
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "alignItems": "center",
                "flexDirection": "row",
                "flexWrap": "wrap",
                "justifyContent": "space-between",
                "opacity": 1,
                "paddingVertical": 8,
              }
            }
          >
            <Text
              style={
                {
                  "color": "#1292B4",
                  "flex": 2,
                  "fontSize": 18,
                  "fontWeight": "400",
                }
              }
            >
              Components
            </Text>
            <Text
              style={
                [
                  {
                    "flex": 3,
                    "fontSize": 18,
                    "fontWeight": "400",
                    "paddingVertical": 16,
                  },
                  {
                    "color": "#444",
                  },
                ]
              }
            >
              The full list of components and APIs inside React Native.
            </Text>
          </View>
          <View
            style={
              [
                {
                  "height": 0.5,
                },
                {
                  "backgroundColor": "#DAE1E7",
                },
              ]
            }
          />
          <View
            accessibilityRole="button"
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "alignItems": "center",
                "flexDirection": "row",
                "flexWrap": "wrap",
                "justifyContent": "space-between",
                "opacity": 1,
                "paddingVertical": 8,
              }
            }
          >
            <Text
              style={
                {
                  "color": "#1292B4",
                  "flex": 2,
                  "fontSize": 18,
                  "fontWeight": "400",
                }
              }
            >
              Navigation
            </Text>
            <Text
              style={
                [
                  {
                    "flex": 3,
                    "fontSize": 18,
                    "fontWeight": "400",
                    "paddingVertical": 16,
                  },
                  {
                    "color": "#444",
                  },
                ]
              }
            >
              How to handle moving between screens inside your application.
            </Text>
          </View>
          <View
            style={
              [
                {
                  "height": 0.5,
                },
                {
                  "backgroundColor": "#DAE1E7",
                },
              ]
            }
          />
          <View
            accessibilityRole="button"
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "alignItems": "center",
                "flexDirection": "row",
                "flexWrap": "wrap",
                "justifyContent": "space-between",
                "opacity": 1,
                "paddingVertical": 8,
              }
            }
          >
            <Text
              style={
                {
                  "color": "#1292B4",
                  "flex": 2,
                  "fontSize": 18,
                  "fontWeight": "400",
                }
              }
            >
              Networking
            </Text>
            <Text
              style={
                [
                  {
                    "flex": 3,
                    "fontSize": 18,
                    "fontWeight": "400",
                    "paddingVertical": 16,
                  },
                  {
                    "color": "#444",
                  },
                ]
              }
            >
              How to use the Fetch API in React Native.
            </Text>
          </View>
          <View
            style={
              [
                {
                  "height": 0.5,
                },
                {
                  "backgroundColor": "#DAE1E7",
                },
              ]
            }
          />
          <View
            accessibilityRole="button"
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "alignItems": "center",
                "flexDirection": "row",
                "flexWrap": "wrap",
                "justifyContent": "space-between",
                "opacity": 1,
                "paddingVertical": 8,
              }
            }
          >
            <Text
              style={
                {
                  "color": "#1292B4",
                  "flex": 2,
                  "fontSize": 18,
                  "fontWeight": "400",
                }
              }
            >
              Debugging
            </Text>
            <Text
              style={
                [
                  {
                    "flex": 3,
                    "fontSize": 18,
                    "fontWeight": "400",
                    "paddingVertical": 16,
                  },
                  {
                    "color": "#444",
                  },
                ]
              }
            >
              Learn about the tools available to debug and inspect your app.
            </Text>
          </View>
          <View
            style={
              [
                {
                  "height": 0.5,
                },
                {
                  "backgroundColor": "#DAE1E7",
                },
              ]
            }
          />
          <View
            accessibilityRole="button"
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "alignItems": "center",
                "flexDirection": "row",
                "flexWrap": "wrap",
                "justifyContent": "space-between",
                "opacity": 1,
                "paddingVertical": 8,
              }
            }
          >
            <Text
              style={
                {
                  "color": "#1292B4",
                  "flex": 2,
                  "fontSize": 18,
                  "fontWeight": "400",
                }
              }
            >
              Help
            </Text>
            <Text
              style={
                [
                  {
                    "flex": 3,
                    "fontSize": 18,
                    "fontWeight": "400",
                    "paddingVertical": 16,
                  },
                  {
                    "color": "#444",
                  },
                ]
              }
            >
              Need more help? There are many other React Native developers who may have the answer.
            </Text>
          </View>
          <View
            style={
              [
                {
                  "height": 0.5,
                },
                {
                  "backgroundColor": "#DAE1E7",
                },
              ]
            }
          />
          <View
            accessibilityRole="button"
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "alignItems": "center",
                "flexDirection": "row",
                "flexWrap": "wrap",
                "justifyContent": "space-between",
                "opacity": 1,
                "paddingVertical": 8,
              }
            }
          >
            <Text
              style={
                {
                  "color": "#1292B4",
                  "flex": 2,
                  "fontSize": 18,
                  "fontWeight": "400",
                }
              }
            >
              Follow us on Twitter
            </Text>
            <Text
              style={
                [
                  {
                    "flex": 3,
                    "fontSize": 18,
                    "fontWeight": "400",
                    "paddingVertical": 16,
                  },
                  {
                    "color": "#444",
                  },
                ]
              }
            >
              Stay in touch with the community, join in on Q&As and more by following React Native on Twitter.
            </Text>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</RCTSafeAreaView>
`;
